source/extensions/filters/http/lua/lua_filter.cc

```diff
@@ -300,21 +345,31 @@ int StreamHandleWrapper::luaRespond(lua_State* state) {
 int StreamHandleWrapper::luaHttpCall(lua_State* state) {
   ASSERT(state_ == State::Running);
 
-  const int async_flag_index = 6;
-  if (!lua_isnone(state, async_flag_index) && !lua_isboolean(state, async_flag_index)) {
-    luaL_error(state, "http call asynchronous flag must be 'true', 'false', or empty");
+  const bool async_flag_is_table = lua_istable(state, AsyncFlagIndex);
+  if (!lua_isnone(state, AsyncFlagIndex) && !lua_isboolean(state, AsyncFlagIndex) &&
+      !async_flag_is_table) {
+    luaL_error(state, "http call asynchronous flag must be 'true', 'false', a table or empty");
+  }
+
+  bool do_async_http_call = lua_toboolean(state, AsyncFlagIndex);
+  auto request_options = Http::AsyncClient::RequestOptions();
+  if (async_flag_is_table) {
+    // The given options table from user may contains "asynchronous" entry that dictates whether the
+    // HTTP call should be done asynchronously or synchronously.
+    do_async_http_call = initializeRequestOptionsFromTable(request_options, state, AsyncFlagIndex);
   }
```

Comment:
nit: to avoid unnecessary `lua_toboolean` when `async_flag_is_table` is `true`.
```
bool do_async_http_call = false;
if (async_flag_is_table) {
  do_async_http_call = ......
} else {
  do_async_http_call = ......
}
```

Reply:
Cool. Thanks for this.